### PR TITLE
tiltfile: use the extension api objects for file loading

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas, @landism,

Please review the following commits I made in branch nicks/extension:

119e28f42cb84f73ac1e41ac5b9eb6a70c0f6510 (2022-02-03 16:59:23 -0500)
tiltfile: use the extension api objects for file loading
Now, extensions are stored in xdg home instead of being vendored in tilt_modules.

You can also change the default extension repo with:

```
v1alpha1.extension_repo(name='default', url='https://github.com/my-org/my-repo', ref='my-tag')
```

at the top of the tiltfile.

This should address the following issues:
https://github.com/tilt-dev/tilt/issues/3188
https://github.com/tilt-dev/tilt/issues/3787
https://github.com/tilt-dev/tilt/issues/4707
https://github.com/tilt-dev/tilt/issues/3426
https://github.com/tilt-dev/tilt/issues/5336
https://github.com/tilt-dev/tilt/issues/5171

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
most of this PR is:
1) moving around deps
2) adding public ForceApply() functions to the reconcilers
3) swapping out the old impl with the new one

(1) and (2) could probably be a separate PR; they just make less sense in isolation.

**Review by kensipe — COMMENTED** [review]
Added a few comments to consider

File: internal/tiltfile/tiltextension/plugin.go [hunk 1]
```diff
@@ -78,25 +88,96 @@ func (e *Plugin) LocalPath(t *starlark.Thread, arg string) (localPath string, er
...
+	}
+
+	typedSet := objSet.GetOrCreateTypedSet(defaultExt)
+	existing, exists := typedSet[extName]
+	if exists {
+		ext := existing.(*v1alpha1.Extension)
+		if ext.ObjectMeta.Annotations == nil {
+			ext.ObjectMeta.Annotations = map[string]string{}
+		}
+		ext.ObjectMeta.Annotations[v1alpha1.AnnotationManagedBy] = "tiltfile.loader"
```
Comment (landism) [inline internal/tiltfile/tiltextension/plugin.go:149]: doesn't matter much, but TIL: https://github.com/tilt-dev/tilt/blob/948bda756a8f0b8e3fc30cec78894333eab9d3df/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go#L199

File: internal/tiltfile/tiltextension/plugin_test.go [hunk 2]
```diff
@@ -122,25 +125,79 @@ printFoo()
...
+
+	f := newExtensionFixture(t)
+	defer f.tearDown()
+
+	f.tiltfile(fmt.Sprintf(`
+v1alpha1.extension_repo(name='default', url='file://%s/my-custom-repo')
+v1alpha1.extension(name='my-extension', repo_name='default', repo_path='my-custom-path')
+
+load("ext://my-extension", "printFoo")
+printFoo()
```
Comment (landism) [inline internal/tiltfile/tiltextension/plugin_test.go:143]: 🎉

**Review by landism — APPROVED** [review]

**Comment by landism** [comment]
I got this output when `tilt up`ing on a Tiltfile w/ uibuttons:
```
ERROR: extension uibutton: extension repo default not found
ERROR: extension uibutton: extension repo default not loaded yet
```

It looks like things worked anyway, but the output led me to believe it didn't.

**Pull request merged**